File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode newHead=new ListNode();
        ListNode node=newHead;
        node.next=head;
        node=node.next;
        ListNode preNode=newHead;
        while(node!=null){
            int count=0;
            int val=node.val;
            ListNode tempNode=node;
            while(tempNode!=null && tempNode.val==val){
                count++;
                tempNode=tempNode.next;
            }
            if(count>1){
                preNode.next=tempNode;
                node=tempNode;
            }else{
                preNode=node;
                node=node.next;
            }
        }
        return newHead.next;
    }
}

File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* node = head;
        ListNode* result = new ListNode();
        ListNode* pre = result;
        while (node != NULL) {
            ListNode* tempNode = node->next;
            bool flag = false;
            while (tempNode != NULL && tempNode->val == node->val) {
                tempNode = tempNode->next;
                flag = true;
            }
            if (!flag) {
                pre->next = node;
                pre = pre->next;
            }
            node = tempNode;
        }
        pre->next=NULL;
        return result->next;
    }
};